project/GenTupleAggregators.scala

```diff
@@ -76,29 +76,32 @@ object MapAggregator {
        |}
     """.stripMargin
 
-    (2 to max).map(i => {
-      val nums = (1 to i)
-      val bs = nums.map("B" + _).mkString(", ")
-      val aggs = nums.map(x => "agg%s: Tuple2[K, %sAggregator[A, B%s, C]]".format(x, aggType, x)).mkString(", ")
-      val prepares = nums.map(x => "agg%s._2.prepare(a)".format(x)).mkString(", ")
-      val semiType = if (isMonoid) "monoid" else "semigroup"
-      val semigroups = nums.map(x => "agg%s._2.%s".format(x, semiType)).mkString(", ")
-      val semigroup = "new Tuple%d%s()(%s)".format(i, semiType.capitalize, semigroups)
-      val present = nums.map(x => "agg%s._1 -> agg%s._2.present(b._%s)".format(x, x, x)).mkString(", ")
-      val tupleBs = "Tuple%d[%s]".format(i, bs)
+    (2 to max).map(aggrCount => {
+      val aggrNums = 1 to aggrCount
 
-      """
-def apply[K, A, %s, C](%s): %sAggregator[A, %s, Map[K, C]] = {
-  new %sAggregator[A, %s, Map[K, C]] {
-    def prepare(a: A) = (%s)
-    val %s = %s
-    def present(b: %s) = Map(%s)
-  }
-}""".format(bs, aggs, aggType, tupleBs,
-            aggType, tupleBs,
-            prepares,
-            semiType, semigroup,
-            tupleBs, present)
+      val inputAggs = aggrNums.map(i => s"agg$i: (K, ${aggType}Aggregator[A, B$i, C])").mkString(", ")
+
+      val semigroup = if (isMonoid) "monoid" else "semigroup"
```

Comment:
I'd prefer not to use word `aggregator` to avoid confusion with an `*Aggregator`.
so either `semigroupType/semigroupField` or a `summer`, other suggestions?